modules/@apostrophecms/image/ui/apos/components/AposMediaManager.vue

```diff
@@ -397,13 +406,33 @@ export default {
       }
       this.currentPage = currentPage;
       this.totalPages = totalPages;
+      // Do not perform extra work if not needed
+      const skipIds = options.loadMoreIfSkipped && !options.loadMore
+        ? new Set(this.items.map(item => item._id))
+        : new Set();
+
+      let skipped = 0;
       for (const item of items) {
+        if (skipIds.has(item._id)) {
+          skipped++;
+          continue;
+        }
         this.items.push(item);
       }
+
+      // In effect when resetting the state (e.g. search) and we want to
+      // load items until the scroll is available.
       if (options.loadMore) {
         this.isLoading = false;
         await this.$nextTick();
         await this.loadUntilScroll(options.lock);
+      } else if (options.loadMoreIfSkipped && skipped > 0) {
```

Comment:
a check on `skipped` would be enough here.

Reply:
See above